engine/src/main/java/org/terasology/rendering/dag/nodes/AlphaRejectBlocksNode.java

```diff
@@ -107,13 +210,77 @@ public void disableWireframe() {
+        cameraPosition = activeCamera.getPosition();
+
+        chunkMaterial.setFloat("swimming", activeCamera.isUnderWater() ? 1.0f : 0.0f, true);
+        chunkMaterial.setFloat3("cameraPosition", cameraPosition.x, cameraPosition.y, cameraPosition.z, true);
+        chunkMaterial.setFloat3("cameraDirection", cameraDir.x, cameraDir.y, cameraDir.z, true);
+        chunkMaterial.setFloat3("cameraParameters", activeCamera.getzNear(), activeCamera.getzFar(), 0.0f, true);
+
+        sunDirection = backdropProvider.getSunDirection(false);
+        chunkMaterial.setFloat3("sunVec", sunDirection.x, sunDirection.y, sunDirection.z, true);
+
+        chunkMaterial.setFloat("time", worldProvider.getTime().getDays(), true);
+
+        // Specific Shader Parameters
+
+        // TODO: This is necessary right now because activateFeature removes all material parameters.
+        // TODO: Remove this explicit binding once we get rid of activateFeature, or find a way to retain parameters through it.
+        chunkMaterial.setInt("textureAtlas", 0, true);
+        chunkMaterial.setInt("textureWater", 1, true);
+        chunkMaterial.setInt("textureLava", 2, true);
+        chunkMaterial.setInt("textureWaterNormal", 3, true);
+        chunkMaterial.setInt("textureWaterNormalAlt", 4, true);
+        chunkMaterial.setInt("textureEffects", 5, true);
+        chunkMaterial.setInt("textureWaterReflection", 6, true);
+        chunkMaterial.setInt("texSceneOpaque", 7, true);
+
+        if (renderingConfig.isNormalMapping()) {
+            if (renderingConfig.isParallaxMapping()) {
+                chunkMaterial.setFloat4("parallaxProperties", parallaxBias, parallaxScale, 0.0f, 0.0f, true);
+            }
+        }
+
+        lightingSettingsFrag.set(0, 0, 0, waterSpecExp);
+        chunkMaterial.setFloat4("lightingSettingsFrag", lightingSettingsFrag, true);
+
+        waterSettingsFrag.set(waterNormalBias, waterRefraction, waterFresnelBias, waterFresnelPow);
+        chunkMaterial.setFloat4("waterSettingsFrag", waterSettingsFrag, true);
+
+        alternativeWaterSettingsFrag.set(waterTint, 0, 0, 0);
+        chunkMaterial.setFloat4("alternativeWaterSettingsFrag", alternativeWaterSettingsFrag, true);
+
+        // TODO: monitor the renderingConfig for changes rather than check every frame
+        if (renderingConfig.isAnimateWater()) {
+            chunkMaterial.setFloat("waveIntensFalloff", waveIntensFalloff, true);
+            chunkMaterial.setFloat("waveSizeFalloff", waveSizeFalloff, true);
+            chunkMaterial.setFloat("waveSize", waveSize, true);
+            chunkMaterial.setFloat("waveSpeedFalloff", waveSpeedFalloff, true);
+            chunkMaterial.setFloat("waveSpeed", waveSpeed, true);
+            chunkMaterial.setFloat("waveIntens", waveIntens, true);
+            chunkMaterial.setFloat("waterOffsetY", waterOffsetY, true);
+            chunkMaterial.setFloat("waveOverallScale", waveOverallScale, true);
+        }
```

Comment:
The other nodes taking advantage of the chunk shader seem to suffer from the same issues I highlighted in this node's code. I won't add the same comments to the others though. 

Perhaps you might want to first aggregate the duplicated code somewhere else, i.e. a common class to inherit from extending AbstractNode.